IntranetGeoIP.php

```diff
@@ -69,118 +71,74 @@ public function activate()
 
     private function copyDataFile()
     {
-        if (! file_exists($this->getDataFilePath()) && file_exists($this->getDataExampleFilePath())) {
+        if (!file_exists($this->getDataFilePath()) && file_exists($this->getDataExampleFilePath())) {
             copy($this->getDataExampleFilePath(), $this->getDataFilePath());
         }
-        
+
         $notification = new Notification('Please edit the file ' . $this->getDataFilePath() . ' and fill in your data');
         $notification->raw = true;
         $notification->context = Notification::CONTEXT_INFO;
         Notification\Manager::notify('IntranetGeoIp_DATA_ERROR', $notification);
-        
+
         return;
     }
 
-    public static function getResult(array $userInfo)
+    /**
+     * @param string|null $visitorIP
+     * @return array
+     */
+    public static function getResult($visitorIP = null)
     {
-        if (!array_key_exists($userInfo['ip'], self::$result)) {
-            
-            self::$result[$userInfo['ip']] = self::getNewResult($userInfo);
+        if (isset($visitorIP)) {
```

Comment:
It's always isset currently, since `IPUtils::binaryToStringIP` returns `0.0.0.0`

`return $ipStr === false ? '0.0.0.0' : $ipStr;`